src/test/java/io/confluent/connect/hdfs/wal/FSWALTest.java

```diff
@@ -72,4 +75,38 @@ public void testTruncatedVersionWalFileRecovery() throws Exception {
     FSWAL wal = new FSWAL("/logs", tp, storage);
     wal.acquireLease();
   }
+
+  @Test
+  public void testAcquireLeaseThrowsException() throws Exception {
+    setUp();
+    HdfsStorage storage = new HdfsStorage(connectorConfig, url);
+    TopicPartition tp = new TopicPartition("mytopic", 123);
+    //stop HDFS cluster so that WALFile.createWriter() throws an exception
+    cluster.shutdown();
+    FSWAL wal = new FSWAL("/logs", tp, storage);
+    int fileSystemCacheSizeBefore = getFileSystemCacheSize();
+    for (int i = 0; i < 10; i++) {
+      try {
+        wal.acquireLease();
```

Comment:
One other question is that you did found out the real issue with Wal file and suggested customer to fix the permissions, don't we need that fix also?

Reply:
It's a generic issue, that could be caused by multiple reasons - lack of HDFS privileges, HDFS is down, etc.
The PR fixes the generic issue of leaking `fs` instance in case there's an exception, different from `RemoteException`.
I also logged a separate ticket to improve permissions validation - https://confluentinc.atlassian.net/browse/CCMSG-1297